guides/authentication-methods/oauth/microsoft.mdx

```diff
@@ -0,0 +1,385 @@
+        alt="Configure optional claim"
+        width={500}
+        style={{ borderRadius: "0.5rem" }}
+    />
+</Frame>
+
+6. Tick the checkbox to set the required Microsoft Graph permissions.
+7. Click `Add`.
+
+<Frame>
+    <img
+        src="/images/oauth-providers/microsoft/set-graph-permissions.png"
+        alt="Set Microsoft Graph permissions for the upn claim"
+        width={500}
+        style={{ borderRadius: "0.5rem" }}
+    />
+</Frame>
+
+### Protection against the nOAuth vulnerability
+
+In 2023 Descope [discovered](https://www.descope.com/blog/post/noauth) a vulnerability resulting
+from misconfigurations of multi-tenant apps that could potentially lead to account takeovers when linking
+accounts with the provider (account linking is activated by default in Hanko). To mitigate this
+vulnerability Microsoft provides the following guidance:
+
+1. Configure the `authenticationBehaviors` setting of your application to disallow unverified email domains. This should
+be the default case for newly created applications, so no further steps should be required.
+
+<Info>
+    If your application was created before the publication of the vulnerability, and the application previously used
+    unverified emails, you might need to manually alter the `authenticationBehaviours` as described
+    [here](https://learn.microsoft.com/en-us/entra/identity-platform/migrate-off-email-claim-authorization#how-do-i-protect-my-application-immediately).
+</Info>
+
+2. Use the optional `xms_edov` token claim to check if an email domain owner has been verified. We recommend implementing
+this additional security measure in case the tokens do not contain a `upn` (e.g. if the claim has not been configured
+or the configuration has been reverted) and decisions have to be based on a users' `email` property. To do so, first open
+your applications `Manifest`.
+
+<Frame>
+    <img
+        src="/images/oauth-providers/microsoft/manifest.png"
+        alt="View app manifest"
+        width={150}
+        style={{ borderRadius: "0.5rem" }}
+    />
+</Frame>
+
+3. Find the `optionalClaims` key and set its value to the following (make a backup of the manifest file, just in case):
+
+```json
+"optionalClaims": {
+		"idToken": [
+			{
+				"name": "xms_edov",
+				"source": null,
+				"essential": false,
+				"additionalProperties": []
+			}
+		]
+	}
+```
+
+<Info>
+    You may notice that after adding the `xms_edov` claim, it is listed in the `Token configuration` section but it is
+    marked with a warning icon claiming that _"[t]his claim is not supported and will not be returned in the token"_.
+    At the time of writing this does not appear to be correct and the claim is in fact returned in the ID token.
+</Info>
+
+
+Hanko per default requires email verification and - with the `xms_edov` claim configured - will reject any attempt at
+connecting third party providers that do not provide a verified email address.
```

Comment:
After reading that section a second time, it felt like that setting the `xms_edov` setting in azure will affect *all* third party providers. Maybe we can clarify that only this specific one is affected